codechecker_common/review_status_handler.py

```diff
@@ -0,0 +1,283 @@
+                        item['report_hash_filter']):
+                continue
+
+            if any(filt in item for filt in
+                   ['filepath_filter', 'checker_filter',
+                    'report_hash_filter']):
+                return SourceReviewStatus(
+                    status=item['review_status'],
+                    message=item['message']
+                    .encode(encoding='utf-8', errors='ignore')
+                    if 'message' in item else b'',
+                    bug_hash=report.report_hash,
+                    in_source=True)
+
+        return None
+
+    def get_review_status_from_source(
+        self,
+        report: Report
+    ) -> Optional[SourceReviewStatus]:
+        """
+        Return the review status based on the source code comment belonging to
+        the given report.
+
+        - Return the review status if it is set in the source code.
+        - If the review status is ambiguous (i.e. multiple source code
+            comments belong to it) then a ValueError exception is raised which
+            contains information about the problem in a string.
+        - If the soure file changed (which means that the source comments may
+            have replaced, changed or removed) then None returns.
+        - If no review status belongs to the report in the source code, then
+            return None.
+        TODO: This function either returns a SourceReviewStatus, or raises an
+        exception or returns None. This is too many things that a caller needs
+        to handle. The reason is that according to the current behaviors,
+        ambiguous comment results a hard error, and changed files result a
+        warning with "unreviewed" status. In the future we could implement a
+        logic where we take the first comment into account in case of ambiguity
+        or return "unreviewed" with a warning, like changed files.
+        """
+        # The original file path is needed here, not the trimmed, because
+        # the source files are extracted as the original file path.
+        # TODO: Should original_path be strip('/') at store?
+        source_file_name = os.path.realpath(os.path.join(
+            self.__source_root, report.file.original_path))
+
+        if source_file_name in report.changed_files:
+            return None
+
+        if os.path.isfile(source_file_name):
+            src_comment_data = self.__parse_codechecker_review_comment(
+                source_file_name, report.line, report.checker_name)
+
+            if len(src_comment_data) == 1:
+                data = src_comment_data[0]
+                status, message = data.status, data.message
+                self.__source_commets[report] = data
+
+                return SourceReviewStatus(
+                    status=status,
+                    message=message.encode('utf-8'),
+                    bug_hash=report.report_hash,
+                    in_source=True)
```

Comment:
Why did you chose to turn `get_review_status_from_source`, but not the assigment to `author` and `date`?

Reply:
Because `ReviewStatusHandler` doesn't know about users. This class is used in client-side commands too, like "CodeChecker parse". Here we don't have a user, only "CodeChecker store" provides a user name. The goal of `ReviewStatusHandler` is to gather the review status from different sources (e.g. source code or config file).